bin/desi_freeze_tilenight

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+
+"""
+Utility script to remove write access (i.e. freeze)
+tiles/cumulative/TILEID/NIGHT directories for which spectro operations have
+signed off as done and ready to release overlapping tiles.  If directory
+already has write permission removed, it is skipped.
+"""
+
+import os, sys, glob
+import argparse
+import subprocess
+
+from astropy.table import Table
+
+from desiutil.log import get_logger
+from desispec.io import specprod_root, findfile
+
+parser = argparse.ArgumentParser(
+        description='Remove write permissions from exposures and '
+                    'cumulative redshift directories for a tile/night, '
+                    'e.g. after accepting that tile as done')
+parser.add_argument('-t', '--tileid', type=int,
+        help='TILEID to freeze')
+parser.add_argument('-n', '--night', type=int,
+        help='NIGHT to freeze')
+parser.add_argument('-p', '--prod', type=int,
+        help = 'Path to input reduction, e.g. '
+               '/global/cfs/cdirs/desi/spectro/redux/daily, '
+               'or simply prod version, like daily. '
+               'Default is $DESI_SPECTRO_REDUX/$SPECPROD.')
+parser.add_argument('--specstatus', type=str,
+        help='tiles-specstatus.ecsv file to use; freeze any tiles with ZDONE=true')
+parser.add_argument('--dry-run', action='store_true',
+        help='print what directories to freeze, without freezing them')
+
+args = parser.parse_args()
+log = get_logger()
+
+if args.prod is None:
+    reduxdir = specprod_root()
+elif args.prod.count("/") == 0:
+    reduxdir = specprod_root(args.prod)
+else:
+    reduxdir = args.prod
+
+if args.specstatus is not None:
+    log.info(f'Reading tiles from {args.specstatus}')
+    tiles = Table.read(args.specstatus)
+    keep = (tiles['SURVEY'] == 'main') & (tiles['ZDONE'] == 'true')
+    tiles = tiles[keep]
+else:
+    tiles = Table(names=('TILEID', 'LASTNIGHT'), dtype=(int, int))
+    tiles.add_row( (args.tileid, args.night) )
+
+#- find and read exposures table for that specprod
+specprod = os.path.basename(reduxdir)
+for expfile in [
+    f'{reduxdir}/tsnr-exposures.fits',
+    f'{reduxdir}/exposuers-{specprod}.fits',
+    ]:
+    if os.path.exists(expfile):
+        log.info(f'Reading exposures from {expfile}')
+        exposures = Table.read(expfile, 1)
```

Comment:
I know the reason for this is that the HDU names changed and this captures both cases without opening things and checking names, but pointing it out since it is technically against our "best practices".

I'm okay with leaving it, though.

Reply:
Okay with leaving this one, so I'll resolve it.